ui/app/src/modules/Preview/QuickCreate.tsx

```diff
@@ -87,32 +86,24 @@ interface QuickCreateMenuButtonProps {
 }
 
 export function QuickCreateMenu(props: QuickCreateMenuProps) {
-  const { anchorEl, onClose, previewItem, onSaveLegacySuccess, onItemClicked } = props;
+  const { anchorEl, onClose, previewItem, onQuickCreateItemSelected, onItemClicked } = props;
   const classes = useStyles({});
   const dispatch = useDispatch();
   const siteId = useActiveSiteId();
   const AUTHORING_BASE = useSelection<string>((state) => state.env.AUTHORING_BASE);
   const defaultFormSrc = `${AUTHORING_BASE}/legacy/form`;
   const [error, setError] = useState<ApiResponse>(null);
   const [quickCreateContentList, setQuickCreateContentList] = useState(null);
-  const [dialogConfig, setDialogConfig] = useSpreadState({
-    open: false,
-    src: defaultFormSrc,
-    type: 'form',
-    inProgress: false
-  });
-
-  const onEmbeddedFormSaveSuccess = ({ data }) => {
-    data.item?.isPage && dispatch(changeCurrentUrl(data.redirectUrl));
-  };
 
   const onNewContentClick = () => {
     onItemClicked?.();
     dispatch(
       showNewContentDialog({
         site: siteId,
         previewItem,
-        compact: false
+        compact: false,
+        // @ts-ignore
+        onContentTypeSelected: onQuickCreateItemSelected ? onQuickCreateItemSelected() : showEditDialog()
```

Comment:
Fix

Reply:
Why would you call `onQuickCreateItemSelected` when the "New Content" option is clicked?

Reply:
added a new prop `onNewContentSelected`